src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,142 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final Object[] EMPTY_ELEMENTDATA = {};
+    private Object[] elementData;
+    private int size;
+
+    public ArrayList(int size) {
+        if (size > 0) {
+            this.elementData = new Object[size];
+            this.size = 0;
+        } else if (size == 0) {
+            this.elementData = EMPTY_ELEMENTDATA;
+            this.size = 0;
+        } else {
+            throw new IllegalArgumentException("Illegal Capacity: " + size);
+        }
```

Comment:
```suggestion
    public ArrayList(int size) {
        if (size > 0) {
            this.elementData = new Object[size];
        } else if (size == 0) {
            this.elementData = EMPTY_ELEMENTDATA;
        } else {
            throw new IllegalArgumentException("Illegal Capacity: " + size);
        }
```

we don't need to initialize size by 0 as it will be initialized by 0 by dafault